web/thesauruses/rust/1/queues_stacks.json

```diff
@@ -0,0 +1,305 @@
+    },
+    "queue_dequeue_return_an_item": {
+      "name": "Dequeue an item (remove from front, return item)",
+      "code": ["deque.pop_front();"]
+    },
+    "queue_dequeue_delete_an_item": {
+      "name": "Dequeue an item (remove from front, don't return)",
+      "not-implemented": true
+    },
+    "queue_peek_at_next_item": {
+      "name": "Look/peek at next available element (from front)",
+      "code": [
+        "deque.front();",
+        "deque.front_mut();   // for a mutable reference"
+      ]
+    },
+    "queue_peek_at_last_item": {
+      "name": "Look/peek at last element (from back)",
+      "code": [
+        "deque.back();",
+        "deque.back_mut();   // for a mutable reference"
+      ]
+    },
+    "queue_duplicate_next_item": {
+      "name": "Duplicate next item",
+      "not-implemented": true
+    },
+    "queue_swap_two_items": {
+      "name": "Swap two items",
+      "code": ["deque.swap(i, j);"],
+      "comment": "Here i and j are indexes of the elements to be swapped."
+    },
+    "queue_get_size": {
+      "name": "Get size (number of items) in the queue",
+      "code": ["deque.len();"]
+    },
+    "queue_get_capacity": {
+      "name": "Check capacity of queue",
+      "code": ["deque.capacity();"]
+    },
+    "queue_export_to_list": {
+      "name": "Export a list of all queue items",
+      "code": ["let vec = Vec::from(deque);"]
+    },
+    "queue_clear_all": {
+      "name": "Clear out all queue items",
+      "code": ["deque.clear();"]
+    },
+    "stack_push_item": {
+      "name": "Push an item (add to top)",
+      "not-implemented": true
+    },
+    "stack_pop_return_item": {
+      "name": "Pop an item (remove from top, return item)",
+      "not-implemented": true
+    },
+    "stack_pop_delete_item": {
+      "name": "Pop an item (remove from top, don't return)",
+      "not-implemented": true
+    },
+    "stack_peek_at_next_item": {
+      "name": "Look/peek at next element (from top)",
+      "not-implemented": true
+    },
+    "stack_peek_at_last_item": {
+      "name": "Look/peek at last element (from bottom)",
+      "not-implemented": true
+    },
+    "stack_duplicate_next_item": {
+      "name": "Duplicate top item",
+      "not-implemented": true
+    },
+    "stack_swap_two_items": {
+      "name": "Swap two items",
+      "not-implemented": true
+    },
+    "stack_get_size": {
+      "name": "Get size (number of items) on the stack",
+      "not-implemented": true
+    },
+    "stack_get_capacity": {
+      "name": "Check capacity of stack",
+      "not-implemented": true
+    },
+    "stack_export_to_list": {
+      "name": "Export a list of all stack items",
+      "not-implemented": true
+    },
+    "stack_clear_all": {
+      "name": "Clear all queue items",
+      "not-implemented": true
+    },
+    "iterate_pointer_data_type": {
+      "name": "Data type of a iterator pointer",
+      "code": ["Iter<'_, T>", "IterMut<'_, T>     // for mutable iterator"]
```

Comment:
Is there an immutable iterator?

Reply:
- `Iter<'_, T>` is an iterator that returns immutable references.
- `IterMut<'_, T>` is an iterator that returns mutable references.

Reply:
Oh, I misread that. That makes sense. 

I'll consider this resolved.